src/Features/CSharp/Portable/Snippets/CSharpConstructorSnippetProvider.cs

```diff
@@ -3,25 +3,38 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Collections.Generic;
 using System.Composition;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.CSharp.Utilities;
+using Microsoft.CodeAnalysis.Editing;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.LanguageService;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
 using Microsoft.CodeAnalysis.Snippets;
 using Microsoft.CodeAnalysis.Snippets.SnippetProviders;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CSharp.Snippets
 {
     [ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
     internal sealed class CSharpConstructorSnippetProvider : AbstractConstructorSnippetProvider
     {
+        private static readonly ISet<SyntaxKind> s_validModifiers = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
```

Comment:
💡 Use `ImmutableSegmentedHashSet<SyntaxKind>`, or even just `ImmutableArray<SyntaxKind>` for one this short

Reply:
The pattern of declaring `s_validModifiers` as `new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)` is very common in roslyn, e.g. all C# keyword recommenders use it ([example](https://github.com/dotnet/roslyn/blob/60eaadf5c5e8782e43ed2455c80380d6243eac8b/src/Features/CSharp/Portable/Completion/KeywordRecommenders/UnsafeKeywordRecommender.cs#L14-L61)). So I think if we change it, we should chage it everywhere at once, which is better to be done in a separate PR